numpy/_core/_add_newdocs.py

```diff
@@ -6131,13 +6122,12 @@
     >>> (arr + arr).dtype.metadata
     mappingproxy({'key': 'value'})
 
-    But if the arrays have different dtype metadata, the metadata may be
-    dropped:
+    If the arrays have different dtype metadata, the first one wins:
```

Comment:
N.B.:  The previous example says "currently", so I think this is fine, but we shouldn't make promises around this.
(the only true safe thing to do, IMO is to never propagate unless the dtype has custom logic)

FWIW, this works for our numerical dtypes, one already can't expect it to work e.g. with strings.

Reply:
I am not sure about the current rules, but the test was failing so I fixed the test and the comment accordingly.